libraries/AP_Proximity/AP_Proximity_ARS408_CAN.h

```diff
@@ -0,0 +1,295 @@
+    } cluster_1_general;
+
+    typedef union cluster_2_quality
+    {
+        struct
+        {
+            uint64_t Cluster_ID : 8;
+            uint64_t Cluster_DistLat_rms1 : 3;
+            uint64_t Cluster_DistLong_rms : 5;
+            uint64_t Cluster_VrelLat_rms1 : 1;
+            uint64_t Cluster_VrelLong_rms : 5;
+            uint64_t Cluster_DistLat_rms2 : 2;
+            uint64_t Cluster_Pdh0 : 3;
+            uint64_t Reserved : 1;
+            uint64_t Cluster_VrelLat_rms2 : 4;
+            uint64_t Cluster_AmbigState : 3;
+            uint64_t Cluster_InvalidState : 5;
+        } data = {};
+        uint8_t raw_data[5];
+    }cluster_2_quality;
+
+
+typedef union radar_cfg {
+  struct {
+    uint64_t RadarCfg_MaxDistance_valid:1;
+    uint64_t RadarCfg_SensorID_valid:1;
+    uint64_t RadarCfg_RadarPower_valid:1;
+    uint64_t RadarCfg_OutputType_valid:1;
+    uint64_t RadarCfg_SendQuality_valid:1;
+    uint64_t RadarCfg_SendExtInfo_valid:1;
+    uint64_t RadarCfg_SortIndex_valid:1;
+    uint64_t RadarCfg_StoreInNVM_valid:1;
+    uint64_t RadarCfg_MaxDistance1:8;
+    uint64_t Reserved:6;
+    uint64_t RadarCfg_MaxDistance2:2;
+    uint64_t Reserved2:8;
+    uint64_t RadarCfg_SensorID:3;
+    uint64_t RadarCfg_OutputType:2;
+    uint64_t RadarCfg_RadarPower:3;
+    uint64_t RadarCfg_CtrlRelay_valid:1;
+    uint64_t RadarCfg_CtrlRelay:1;
+    uint64_t RadarCfg_SendQuality:1;
+    uint64_t RadarCfg_SendExtInfo:1;
+    uint64_t RadarCfg_SortIndex:3;
+    uint64_t RadarCfg_StoreInNVM:1;
+    uint64_t RadarCfg_RCS_Threshold_valid:1;
+    uint64_t RadarCfg_RCS_Threshold:3;
+    uint64_t Reserved3:4;
+    uint64_t Reserved4:8;
+  } data = {};
+
+  uint8_t raw_data[8];
+} radar_cfg;
+
+    // update state
+    void update(void) override;
+
+    // get maximum and minimum distances (in meters) of sensor
+    float distance_max() const override;
+    float distance_min() const override;
+
+    // handler for incoming frames
+    void handle_frame(AP_HAL::CANFrame &frame) override;
+
+protected:
+    void send_speed_message();
+    void send_yaw_rate_message();
+    bool get_forward_speed(float &speed) const;
+    bool set_max_distance(uint64_t distance, bool valid = true);
+
+    bool set_sensor_id(int id, bool valid = true);
+    bool set_radar_power(int power, bool valid = true);
+    bool set_output_type(int output_type, bool valid = true);
+    void set_send_quality(bool quality, bool valid = true);
+    void set_send_ext_info(bool send_ext, bool valid = true);
+    bool set_sort_index(int sort_index, bool valid = true);
+    void set_ctrl_relay_cfg(bool ctrl_relay, bool valid = true);
+    void set_store_in_nvm(bool store_in_nvm, bool valid = true);
+    bool set_rcs_threshold(int rcs_threshold, bool valid = true);
+
+private:
+    object_0_status obj_status_;
+    std::vector<object_1_general> obj_general_list_;
+    std::vector<object_2_quality> obj_quality_list_;
+    std::vector<object_3_extended> obj_extended_list_;
```

Comment:
No std::vector allowed, sorry.  Or anything from std::, in fact.  You'll need to find another way...